controllers/azuremachine_controller.go

```diff
@@ -284,6 +284,15 @@ func (amr *AzureMachineReconciler) reconcileNormal(ctx context.Context, machineS
 		return reconcile.Result{}, errors.Wrap(err, "failed to init machine scope cache")
 	}
 
+	// If a node is unhealthy, trigger a MachineHealthCheck to replace the node.
```

Comment:
Setting up the MachineHealthCheck is an extra step that you have to add in the Spec. This [page](https://cluster-api.sigs.k8s.io/tasks/automated-machine-management/healthchecking.html?highlight=healthchecking) in the CAPI book gives a good overview on how it works.

Reply:
This comment might be a bit confusing since the code itself isn't triggering a MachineHealthCheck and that needs to be set up separately as you point out. This also isn't checking for the node health, but the VM conditions. How about something like:

```suggestion
	// Mark the AzureMachine as failed if any conditions are false.
```